include/archi/ibex/mhpm.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2020 ETH Zurich and University of Bologna
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _ARCHI_IBEX_MHPM_H
+#define _ARCHI_IBEX_MHPM_H
+
+/*
+ * Bit definitions for Performance counters mode registers
+ *
+ */
+
+#define IBEX_MHPMCOUNTER_BASE 0xB00
+// Currently the additional register for more perf. counting is not yet implemented
+
+// ibex performance counters
+#define IBEX_NumCycles        0  /* Number of cycles */
+#define IBEX_NumInstrRet      2  /* Number of instructions retired */
+#define IBEX_NumCyclesLSU     3  /* Number of cycles waiting for data memory */
+#define IBEX_NumCyclesIF      4  /* Cycles waiting for instruction fetches, i.e., number of instructions wasted due to non-ideal caching */
+#define IBEX_NumLoads         5  /* Number of data memory loads. Misaligned accesses are counted as two accesses */
+#define IBEX_NumStores        6  /* Number of data memory stores. Misaligned accesses are counted as two accesses */
+#define IBEX_NumJumps         7  /* Number of unconditional jumps (j, jal, jr, jalr) */
+#define IBEX_NumBranches      8  /* Number of branches (conditional) */
+#define IBEX_NumBranchesTaken 9  /* Number of taken branches (conditional) */
+#define IBEX_NumInstrRetC     10 /* Number of compressed instructions retired */
+#define IBEX_NumCyclesMulWait 11 /* Cycles waiting for multiply to complete */
+#define IBEX_NumCyclesDivWait 12 /* Cycles waiting for divide to complete */
```

Comment:
Minor nit: Macros should be all uppercase

Reply:
I used the names from the [ibex docs](https://ibex-core.readthedocs.io/en/latest/03_reference/performance_counters.html), but I will update this.